src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -24,6 +27,10 @@ public class SystemNetHttpClient : IHttpClient
 
         private readonly RequestTelemetry requestTelemetry = new RequestTelemetry();
 
+        private readonly object randLock = new object();
+
+        private readonly Random rand = new Random();
```

Comment:
This random number generator is used to generate the jitter in retry delays. Microsoft recommends creating a single generator instance rather than creating a new instance every time, but also warns that `Random` is not thread-safe, so a lock is needed. Cf. https://docs.microsoft.com/en-us/dotnet/api/system.random?view=netframework-4.7.2#Multiple.

Reply:
Nice research. (And wow, a thread-safe convenient wrapper for `Rand` seems like a pretty obvious improvement for a future version of .NET.)